std/math.d

```diff
@@ -4103,9 +4103,55 @@ long lrint(real x) @trusted pure nothrow @nogc
 
             return sign ? -result : result;
         }
+        else static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            long result;
+
+            // Rounding limit when casting from real(128-bit) to ulong.
+            enum real OF = 0x1P+112;
+
+            ulong* vl = cast(ulong*)(&x);
+
+            enum EXPSHIFT_ULONG = 48;
+            enum MANTISSE = real.mant_dig - 1;
```

Comment:
Are these strictly necessary?  I don't think anyone would mind if you just use the numbers directly in the implementation, as they are very specific to this function.

The `MANTISSE` should always be `112`, no?